rosbag2_transport/test/rosbag2_transport/test_record_all_use_sim_time.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "rosbag2_test_common/publication_manager.hpp"
+#include "rosbag2_test_common/wait_for.hpp"
+
+#include "rosbag2_transport/recorder.hpp"
+
+#include "test_msgs/msg/basic_types.hpp"
+#include "test_msgs/message_fixtures.hpp"
+
+#include "record_integration_fixture.hpp"
+#include "rosgraph_msgs/msg/clock.hpp"
+
+std::vector<rosgraph_msgs::msg::Clock::SharedPtr> get_clock_messages()
+{
+  std::vector<rosgraph_msgs::msg::Clock::SharedPtr> messages;
+  for (int i = 0; i < 5; ++i) {
+    auto msg = std::make_shared<rosgraph_msgs::msg::Clock>();
+    msg->clock.sec = 1234567890;
+    msg->clock.nanosec = 123456789;
+    messages.push_back(msg);
+  }
+  return messages;
+}
```

Comment:
We actually don't need to prepare vector of messages. We need to prepare only one message. Publication manager will repeat it anyway and will not operate with vector.

Reply:
Publication manager is very useful. I have modified it as suggested.
ecd0fa6